main.tf

```diff
@@ -351,10 +351,10 @@ resource "aws_iam_role_policy_attachment" "cloudwatch_to_fh" {
 }
 
 resource "aws_cloudwatch_log_subscription_filter" "cloudwatch_log_filter" {
+  count = var.cloudwatch_log_filter_name != "" ? 1 : 0
```

Comment:
AWS now supports up to two subscription filters per log group.  So, the code as it stands right now creates one (1) subscription filter.  If you would like a second filter, why not just create another `aws_cloudwatch_log_subscription_filter` resource in your code that wraps this module to create your second one?  

The change you are proposing here will prevent any `aws_cloudwatch_log_subscription_filter` resources from being created if you leave `var.cloudwatch_log_filter_name` blank which would cause your log group not to be subscribed to any destination ARN and your logs would not get shipped anywhere.